dirsrvtests/tests/suites/disk_monitoring/disk_monitoring_divide_test.py

```diff
@@ -0,0 +1,102 @@
+# --- BEGIN COPYRIGHT BLOCK ---
+# Copyright (C) 2021 Red Hat, Inc.
+# All rights reserved.
+#
+# License: GPL (version 3 or any later version).
+# See LICENSE for details.
+# --- END COPYRIGHT BLOCK ---
+#
+import time
+import subprocess
+import pytest
+from lib389.tasks import *
+from lib389.utils import *
+from lib389.topologies import topology_st
+from lib389._mapped_object import DSLdapObjects
+
+pytestmark = pytest.mark.tier2
+
+logging.getLogger(__name__).setLevel(logging.DEBUG)
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="function")
+def create_dummy_mount(topology_st, request):
+    cmds = ['setenforce 0',
+            'mkdir /var/log/dirsrv/slapd-{}/tmp'.format(topology_st.standalone.serverid),
+            'mount -t tmpfs tmpfs /var/log/dirsrv/slapd-{}/tmp -o size=0'.format(topology_st.standalone.serverid),
+            'chown dirsrv: /var/log/dirsrv/slapd-{}/tmp'.format(topology_st.standalone.serverid)]
+
+    log.info('Create dummy mount')
+    for cmd in cmds:
+        log.info('Command used : %s' % cmd)
+        subprocess.Popen(cmd, shell=True)
+
+    def fin():
+        cmds = ['umount /var/log/dirsrv/slapd-{}/tmp'.format(topology_st.standalone.serverid),
+                'setenforce 1']
+
+        for cmd in cmds:
+            log.info('Command used : %s' % cmds)
+            subprocess.Popen(cmd, shell=True)
+
+    request.addfinalizer(fin)
+
+
+@pytest.fixture(scope="function")
+def change_config(topology_st):
+    topology_st.standalone.config.set('nsslapd-disk-monitoring', 'on')
+    topology_st.standalone.config.set('nsslapd-disk-monitoring-readonly-on-threshold', 'on')
+
+
+@pytest.mark.ds4414
+@pytest.mark.bz1890118
+@pytest.mark.skipif(ds_is_older("1.4.3.16"), reason="Might fail because of bz1890118")
+def test_produce_division_by_zero(topology_st, create_dummy_mount, change_config):
+    """Test dirsrv will not crash when division by zero occurs
+
+    :id: 51b11093-8851-41bd-86cb-217b1a3339c7
+    :customerscenario: True
+    :setup: Standalone
+    :steps:
+        1. Turn on disk monitoring
+        2. Go below the threshold
+        3. Check DS is up and not entering shutdown mode
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+    """
+
+    standalone = topology_st.standalone
+
+    log.info('Check search works before changing the nsslapd-auditlog attribute')
+    try:
+        DSLdapObjects(topology_st.standalone, basedn='cn=disk space,cn=monitor').filter("(objectclass=*)", scope=0)
```

Comment:
We have an object that can be used for that - https://github.com/389ds/389-ds-base/blob/master/src/lib389/lib389/monitor.py#L353 - MonitorDiskSpace.

Could you please try if it'll work with '**filter**' function?

Reply:
I tried the suggested object but it doesn't have filter attribute.

Reply:
Ah, right, MonitorDiskSpace is a singular object (and only DSLdapObjects has the filter method).